shell/platform/android/external_view_embedder/external_view_embedder.cc

```diff
@@ -9,32 +9,47 @@
 namespace flutter {
 
 AndroidExternalViewEmbedder::AndroidExternalViewEmbedder(
-    std::shared_ptr<PlatformViewAndroidJNI> jni_facade)
-    : ExternalViewEmbedder(), jni_facade_(jni_facade) {}
+    std::shared_ptr<AndroidContext> android_context,
+    std::shared_ptr<PlatformViewAndroidJNI> jni_facade,
+    const AndroidSurface::Factory& surface_factory)
+    : ExternalViewEmbedder(),
+      android_context_(std::move(android_context)),
+      jni_facade_(std::move(jni_facade)),
+      surface_factory_(surface_factory),
+      surface_pool_(std::make_unique<SurfacePool>()) {}
 
 // |ExternalViewEmbedder|
 void AndroidExternalViewEmbedder::PrerollCompositeEmbeddedView(
     int view_id,
     std::unique_ptr<EmbeddedViewParams> params) {
-  // TODO(egarciad): Implement hybrid composition.
-  // https://github.com/flutter/flutter/issues/55270
   TRACE_EVENT0("flutter",
                "AndroidExternalViewEmbedder::PrerollCompositeEmbeddedView");
+
+  auto rtree_factory = RTreeFactory();
+  view_rtrees_[view_id] = rtree_factory.getInstance();
   picture_recorders_[view_id] = std::make_unique<SkPictureRecorder>();
-  picture_recorders_[view_id]->beginRecording(SkRect::Make(frame_size_));
+  picture_recorders_[view_id]->beginRecording(SkRect::Make(frame_size_),
```

Comment:
A general comment about technique when working with maps: Unless you actually want to insert into the map, prefer `at()` if you know the key if valid or  `find` with an iterator check if you are not sure. Using `operator[]` for access sometimes leads to the insertion of a default constructed value for invalid keys which makes issue due to invalid keys harder to diagnose.

IMO, all these can be`.at()`.

Reply:
Done. Thanks for the suggestion